jwst/extract_2d/grisms.py

```diff
@@ -476,7 +476,10 @@ def extract_grism_objects(input_model,
                 # The overall subarray offset is recorded in model.meta.subarray.
                 # nslit = obj.sid - 1  # catalog id starts at zero
                 new_slit.name = "{0}".format(obj.sid)
-                new_slit.source_type = 'UNKNOWN'
+                if obj.is_star:
+                    new_slit.is_star = True
```

Comment:
I don't think any logic is needed here to make a choice of what to set, but instead simply set `new_slit.is_star = obj.is_star`, in order to carry along the value of `is_star` to the `srctype` step, where it will set the appropriate value of source_type.

Reply:
yes I meant to remove the settling of source type. Some how my push did not include this update